### PR TITLE
Add optional suffix to PathBuilderFactory

### LINKED ISSUE #1891: Using alternative entity alias when it is looks like a part of HQL keywords
For example QueryDSL generates HQL query for delete entity Group with Q-Class QGroup as:
```
delete from Group group
where group = ?1
```
And Hibernate throws an exception:

```
org.hibernate.hql.internal.ast.QuerySyntaxException: unexpected token: group near line 1, column 55 [delete from com.lukoil.wellinfo.dao.model.admin.Group group
where group = ?1]
```

Looks like Hibernate decide that "group" is part of  keyword "group by" instead alias.
May be QueryDSL should be generate another alias name for entity if it looks like a part of HQL keywords?

### PR DESCRIPTION
**Pull request by timowest**
Fixes #1891

### CODE REVIEW

**Comment by Shredder121** [comment]
Hm, seems to work.
Is there anything else you want to add?

**Comment by timowest** [comment]
I'll comment in the issue.

File: querydsl-core/src/main/java/com/querydsl/core/types/dsl/PathBuilderFactory.java [hunk 1]
```diff
@@ -38,10 +48,14 @@
...
         PathBuilder<T> rv = (PathBuilder<T>) paths.get(type);
         if (rv == null) {
-            rv = new PathBuilder<T>(type, CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_CAMEL, type.getSimpleName()));
+            rv = new PathBuilder<T>(type, variableName(type));
             paths.put(type, rv);
         }
         return rv;
     }
 
+    private <T> String variableName(Class<T> type) {
```
Comment (Shredder121) [inline querydsl-core/src/main/java/com/querydsl/core/types/dsl/PathBuilderFactory.java]: I don't think this type parameter is necessary.

Comment (timowest) [inline querydsl-core/src/main/java/com/querydsl/core/types/dsl/PathBuilderFactory.java]: good point

**Comment by Shredder121** [comment]
Looks fine to me.

**Pull request merged**